Client/Frontend/Browser/BrowserViewController/BrowserViewController+WKNavigationDelegate.swift

```diff
@@ -157,6 +159,37 @@ extension BrowserViewController: WKNavigationDelegate {
             return
         }
         #endif
+        
+        if !navigationAction.isInterstitial {
+            if Preferences.Shields.googleSafeBrowsing.value {
+                var safeBrowsingResult: SafeBrowsingResult = .safe
+                let semaphore = DispatchSemaphore(value: 0)
+                SafeBrowsingClient.shared.find(url.hashPrefixes()) { result, error in
+                    defer { semaphore.signal() }
+                    
+                    if let error = error {
+                        log.error(error)
+                        safeBrowsingResult = result
+                        return
+                    }
+                    
+                    safeBrowsingResult = result
+                }
+                
+                if semaphore.wait(timeout: .now() + .seconds(10)) == .success {
```

Comment:
Won't this block the main queue & UI? Also aren't semaphores discouraged over using `DispatchGroup` now?

Reply:
DispatchGroup I thought was for when multiple things are happening at once whereas semaphore is just for synchronization of threads.

I do it like this in order to determine whether or not the page needs to be blocked. I don't know of another way without duplicating a LOT of code.

Reply:
Fixed.